core/utils/svg_paths.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @fileoverview Methods for creating parts of SVG path strings.  See
+ * developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+/**
+ * @name Blockly.utils.Paths
+ * @namespace
+ */
+goog.provide('Blockly.utils.Paths');
+
+
+/**
+ * Create a string representing the given x, y pair.  It does not matter whether
+ * the coordinate is relative or absolute.  The result has leading
+ * and trailing spaces, and separates the x and y coordinates with a comma but
+ * no space.
+ * @param {number} x The x coordinate.
+ * @param {number} y The y coordinate.
+ * @return {string} A string of the format ' x,y '
+ * @public
+ */
+Blockly.utils.Paths.point = function(x, y) {
+  return ' ' + x + ',' + y + ' ';
+};
+/**
+ * Draw a curbic or quadratic curve.  See
+ * developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d#Cubic_B%C3%A9zier_Curve
+ * These coordinates are unitless and hence in the user coordinate system.
+ * @param {string} command The command to use.
+ *     Should be one of: c, C, s, S, q, Q.
+ * @param {string} points  An array containing all of the points to pass to the
+ *     curve command, in order.  The points are represented as strings of the
+ *     format ' x, y '.
+ * @return {string} A string defining one or more Bezier curves.  See the MDN
+ *     documentation for exact format.
+ * @public
+ */
+Blockly.utils.Paths.curve = function(command, points) {
+  return command + points.join('');
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * The coordinates are absolute.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} x The absolute x coordinate.
+ * @param {number} y The absolute y coordinate.
+ * @return {string} A string of the format ' M x,y '
+ * @public
+ */
+Blockly.utils.Paths.moveTo = function(x, y) {
+  return ' M ' + x + ',' + y + ' ';
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * Coordinates are relative.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' m dx,dy '
+ * @public
+ */
+Blockly.utils.Paths.moveBy = function(dx, dy) {
+  return ' m ' + dx + ',' + dy + ' ';
+};
+
+/**
+ * Draw a line from the current point to the end point, which is the current
+ * point shifted by dx along the x-axis and dy along the y-axis.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' L dx,dy '
```

Comment:
Lowercase the 'L' in the JSDoc since that means x/y not dx/dy.

Reply:
Done.  Good catch.